### PR TITLE
Chunk upload file version

### PR DESCRIPTION
**Pull request by carycheng77**
Updated functionality for upload file version using session
Test case for UploadFileVersionUsingSessionAsync()
Updated BoxFile model to include BoxFileVersion

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @carycheng77 has signed the CLA. Thanks for the pull request!

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -318,7 +322,7 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
         /// <param name="commitSessionUrl">Commit URL returned in the Create Session response.</param>
         /// <param name="sha">The message digest of the complete file, formatted as specified by RFC 3230.</param>
         /// <param name="sessionPartsInfo">Parts info for the uploaded parts.</param>
-        /// <returns> The complete BoxFile object. </returns>
+        /// <returns> A collection containing the complete BoxFile object(s). </returns>
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: BoxFile object

File: Box.V2/Managers/BoxFilesManager.cs [hunk 2]
```diff
@@ -455,16 +522,32 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
...
+                await Retry.ExecuteAsync(
+                    async () =>
+                        await CommitFileVersionSessionAsync(commitUri, fullFileSha1,
+                            sessionPartsForCommit),
+                        TimeSpan.FromMilliseconds(retryInterval), retryCount);
+                return versionResponse;
+            }
         }
 
+        public delegate void uploadUsingSessionDelegateAsync();
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: What's this?

**Comment by coolboy** [comment]
#373

**Pull request merged**